presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -849,4 +856,28 @@ private static String typedNull(String prestoType)
     {
         return format("CAST(NULL AS %s)", prestoType);
     }
+
+    private static List<ColumnInfo> getDescribeOutputColumnInfoList(ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
+        while (resultSet.next()) {
+            String columnName = resultSet.getString("Column Name");
+            String catalog = resultSet.getString("Catalog");
+            String schema = resultSet.getString("Schema");
+            String table = resultSet.getString("Table");
+            ClientTypeSignature clientTypeSignature = new ClientTypeSignature(resultSet.getString("Type"));
```

Comment:
In my understanding, using ClientTypeSignature here isn't correct. If the type contains decimal or something other types, it will throw below exception.
```
java.lang.IllegalArgumentException: Bad characters in rawType type: xxx
```

Reply:
@ebyhr thank you for pointing out, will take a look

Reply:
@ebyhr Just re-think about your words. Seems we need to create `ClientTypeSignature` given an arbitrary `Type` string. To do this, I think I will do it recursively i.e. DFS parse the string and create `ClientTypeSignature`. What do you think?